live/client.py

```diff
@@ -3,6 +3,12 @@
 import time
 from urllib.parse import urlparse
 import hashlib
+import xml.etree.ElementTree as ET
+
+
+class InvalidHTTPResponse(Exception):
+    """Execption raised by 'requests' returned with invalid http status code"""
+    pass
```

Comment:
When I'm building a client to an API like this I like to have a base exception, something like:

```python
class ElementalException(Exception):
    """Base exception for all exceptions ElementalLive client could raise
    pass
```

Then I can define more specific ones extending from it: 

```python
class InvalidHTTPResponse(ElementalException):
    """Raised when a HTTP request sent to Elemental fail to complete or 
       returned an invalid status code
    """
    pass
```

this way users of our library can choose to handle all exceptions or be more specific and handle specific ones. 

You're using this behavior from `requests` for example. You're handling `requests.exceptions.RequestException` which is the base class for **all** exceptions raised by `requests`:
https://github.com/kennethreitz/requests/blob/master/requests/exceptions.py

That means if you have a `except RequestException` you will catch every exception that is a subclass of `RequestException`.

The idea is for our library to hide the fact that we use `requests` at all. For clients using `ElementalLive` it doesn't matter _how_ the request is being made, only that it was made and succeeded or not.

For example, did you know [`requests` uses `urllib3`](https://github.com/kennethreitz/requests/blob/master/requests/adapters.py#L14-L29)? See their [`send`](https://github.com/kennethreitz/requests/blob/master/requests/adapters.py#L394) method, note how it handles all urllib3 exceptions and raise `requests` exceptions instead. We want to do the same here

Reply:
Hi Igor, these are super helpful and specific advices. Thank you for being so patient! Learned a lot from you!!! Thanks